### PR TITLE
refactor(*) use pdk and remove BasePlugin inheritance

### PR DESCRIPTION
**Pull request by darrenjennings**
This plugin is going to be a bundled with Kong. This work is from PR feedback on https://github.com/Kong/kong/pull/4685

### CODE REVIEW

File: kong/plugins/session/access.lua [hunk 1]
```diff
@@ -52,20 +52,20 @@ function _M.execute(conf)
...
-    ngx.log(ngx.DEBUG, "No consumer, destroying session")
+    kong.log.debug("No consumer, destroying session")
     return s:destroy()
   end
 
   s:start()
 
   set_consumer(consumer, credential)
-  ngx.ctx.authenticated_session = s
+  kong.ctx.authenticated_session = s
```
Comment (bungle) [inline kong/plugins/session/access.lua]: AFAIK it should be `kong.ctx.shared.authenticated_session` or `kong.ctx.plugin.authenticated_session` depending whether the value is used to for this plugin or shared with other plugins.

File: kong/plugins/session/handler.lua [hunk 2]
```diff
@@ -2,26 +2,24 @@ local BasePlugin = require "kong.plugins.base_plugin"
 local access = require "kong.plugins.session.access"
 local session = require "kong.plugins.session.session"
 
+local kong = kong
 
 -- Grab pluginname from module name
 local plugin_name = ({...})[1]:match("^kong%.plugins%.([^%.]+)")
 
-local KongSessionHandler = BasePlugin:extend()
+local KongSessionHandler = {}
```
Comment (bungle) [inline kong/plugins/session/handler.lua]: You can perhaps move `PRIORITY` and `VERSION` inside `table` constructor:
```
local KongSessionHandler = {
  VERSION  = "1.0.0".
  PRIORITY = 1900,
}
```

File: kong/plugins/session/handler.lua [hunk 3]
```diff
@@ -2,26 +2,24 @@ local BasePlugin = require "kong.plugins.base_plugin"
...
 
 KongSessionHandler.PRIORITY = 1900
 KongSessionHandler.VERSION = "1.0.0"
 
-function KongSessionHandler:new()
-  KongSessionHandler.super.new(self, plugin_name)
-end
 
 function KongSessionHandler:header_filter(conf)
   KongSessionHandler.super.header_filter(self)
```
Comment (bungle) [inline kong/plugins/session/handler.lua]: This should be removed.

File: kong/plugins/session/access.lua (hunk 1 above)
Comment (darrenjennings) [reply kong/plugins/session/access.lua]: We have had plugins destroy a session based on some plugin chain so changed all to `shared`.

File: kong/plugins/session/handler.lua [hunk 4]
```diff
@@ -2,26 +2,23 @@ local BasePlugin = require "kong.plugins.base_plugin"
...
 
+local kong = kong
 
 -- Grab pluginname from module name
 local plugin_name = ({...})[1]:match("^kong%.plugins%.([^%.]+)")
 
-local KongSessionHandler = BasePlugin:extend()
+local KongSessionHandler = {
+  PRIORITY = 1900,
+  VERSION = "1.0.0",
```
Comment (thibaultcha) [inline kong/plugins/session/handler.lua]: It seems like the current version of this plugin in its rockspec is 1.0.2? https://github.com/Kong/kong-plugin-session/blob/master/kong-plugin-session-1.0.2-3.rockspec#L9

File: kong/plugins/session/storage/kong.lua [hunk 5]
```diff
@@ -91,15 +91,15 @@ function kong_storage:insert_session(sid, data, expires)
   }, { ttl = self.lifetime })
 
   if err then
-    ngx.log(ngx.ERR, "Error inserting session: ", err)
+    kong.log.err("Error inserting session: ", err)
```
Comment (thibaultcha) [inline kong/plugins/session/storage/kong.lua]: Just a note, not a blocker for this PR:

1. At runtime (in Kong's proxy and Admin API, for instance), we never capitalize error messages
2. We typically phrase errors like so: `"could not do X:", err`, or `"failed to do X:", err`. Mentioning "Error" here creates duplication in the log which already has an `[error]` prefix indicating its log level.

File: kong/plugins/session/handler.lua (hunk 4 above)
Comment (darrenjennings) [reply kong/plugins/session/handler.lua]: I had changed this to match the rockspec but was told they were not the same thing. Still a bit confused on what versions to sync.
https://kongstrong.slack.com/archives/CEDRYUABS/p1545855566003000

File: kong/plugins/session/storage/kong.lua (hunk 5 above)
Comment (darrenjennings) [reply kong/plugins/session/storage/kong.lua]: oh good note, makes sense

File: kong/plugins/session/handler.lua (hunk 4 above)
Comment (thibaultcha) [reply kong/plugins/session/handler.lua]: Hmmm I don't see any reason for not syncing the plugin's and the rockspec's version (afaik, all plugins have been doing so, and all Lua libraries that I know of follow a similar pattern). The revision number is different, and only applies to the rockspec itself.

Typically:

- Plugin/Library gets a version number: `"x.y.z"`
- The rockspec points to the `x.y.z` tag to refer to the version it needs to install
- The rockspec's version is `"x.y.z-n"`, where `n` is the revision number for the rockspec. It is common that a rockspec needs an update (fix/feature) that doesn't affect the plugin/library itself, and in such cases, the maintainer bumps the revision number to `n+1`.

**Pull request merged**